File: app/src/main/java/kr/kro/awesometic/plankhelper/statistics/calendar/CalendarContract.java
package kr.kro.awesometic.plankhelper.statistics.calendar;

import kr.kro.awesometic.plankhelper.BasePresenter;
import kr.kro.awesometic.plankhelper.BaseView;

/**
 * Created by Awesometic on 2017-05-17.
 */

public interface CalendarContract {

    interface View extends BaseView<Presenter> {

        void showLoading();
        void showCalendar();

        Object getActivityContext();
        void setRecyclerViewAdapter(Object recyclerViewAdapter);

        void setCalendarMinDate(int year, int month, int date);
        void setCalendarMaxDate(int year, int month, int date);

        void addCalendarDecorator(Object decorator);

        void bindViewsFromViewHolder();
    }

    interface Presenter extends BasePresenter {

        void bindViewsFromViewHolderToFrag();

        void onMonthChanged(Object calendarDay);
    }
}

File: app/src/main/java/kr/kro/awesometic/plankhelper/statistics/calendar/CalendarContract.kt
package kr.kro.awesometic.plankhelper.statistics.calendar

import kr.kro.awesometic.plankhelper.BasePresenter
import kr.kro.awesometic.plankhelper.BaseView

/**
 * Created by Awesometic on 2017-05-17.
 */

interface CalendarContract {

    interface View : BaseView<Presenter> {

        val applicationContext: Any

        fun showLoading()
        fun showCalendar()
        fun setRecyclerViewAdapter(recyclerViewAdapter: Any)

        fun bindViewsFromViewHolder()
    }

    interface Presenter : BasePresenter {

        fun bindViewsFromViewHolderToFrag()
    }
}
